index.js

```diff
@@ -190,16 +228,38 @@ function redisStore(args) {
     options = options || {};
 
     var ttl = (options.ttl || options.ttl === 0) ? options.ttl : redisOptions.ttl;
+    var gzip = (options.gzip || options.gzip === false) ? options.gzip : redisOptions.gzip;
+    if (gzip === true) {
+      gzip = gzipDefault;
+    }
 
     connect(function(err, conn) {
       if (err) {
         return cb && cb(err);
       }
       var val = JSON.stringify(value) || '"undefined"';
-      if (ttl) {
-        conn.setex(key, ttl, val, handleResponse(conn, cb));
-      } else {
-        conn.set(key, val, handleResponse(conn, cb));
+
+      if (gzip) {
+        zlib.gzip(val, gzip, function (gzErr, gzVal) {
+          if (gzErr) {
+            return cb && cb(gzErr);
+          }
+
+          if (ttl) {
+            conn.setex(key, ttl, gzVal, handleResponse(conn, cb));
+          }
+          else {
+            conn.set(key, gzVal, handleResponse(conn, cb));
+          }
+        });
+      }
+      else {
+        if (ttl) {
+          conn.setex(key, ttl, val, handleResponse(conn, cb));
+        }
+        else {
+          conn.set(key, val, handleResponse(conn, cb));
```

Comment:
Ok. Makes sense. I'm still a little confused, though. Sorry! 

Right now, there is `self.set()`. In order to refactor the duplicate code in lines `248-262`, there will need to be a new function that basically does this:
```js
function newSet(key, value, ttl, cb) {
  if (typeof ttl === 'function') {
    cb = ttl;
    ttl = '';
  }

  if (ttl) {
    conn.setex(key, ttl, value, handleResponse(conn, cb));
  }
  else {
    conn.set(key, value, handleResponse(conn, cb));
  }
}
```
Then lines `248-253` and `257-262` would each be replaced by:
```js
  newSet(key, gzVal, ttl, cb);
// or
  newSet(key, val, ttl, cb);
```
Is this what you are thinking of? If so, what would you like to call `newSet`? It's basically a helper function for `self.set`.